src/persistence/src/persistent_db.rs

```diff
@@ -46,9 +49,86 @@ impl PersistentDb {
         PersistentDb {
             db_ref: None,
             cf: None,
+            tx: None,
             memory_db: Some(HashMap::new()),
         }
     }
+
+    /// Commits the pending transactions to the db
+    fn flush(&self) {
```

Comment:
In Rust, things move around. Which means that passing `tx` to `write()` does not work.

Instead you need to clone the transaction and then pass it to the `write()` function like so:
```rust
/// Commits the pending transactions to the db.
fn flush(&mut self) {
    if let (Some(db_ref), Some(tx)) = (&self.db_ref, &self.tx) {
        db_ref.write(tx.clone()).unwrap();
    }

    // Wipe pending state
    self.wipe()
}
```